spec/models/user_notification_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe UserNotification, type: :model do
+  describe 'Create notifications' do
+    let(:user) { create(:user) }
+    let(:activity) do
+      create(:activity, actor: user, object: user, event: :tested)
+    end
+
+    context 'when creating an email notification' do
+      let(:email_notification) do
+        build(:user_notification, activity: activity, notification_type: :email)
+      end
+
+      it 'sends out email' do
+        pending 'implement send_email method'
+        expect { email_notification.save }.to change { ActionMailer::Base.deliveries.count }.by(1)
+      end
+    end
+
+    context 'when creating popup notification' do
+      let(:popup_notification) do
```

Comment:
Actually do you need to test like this ? I think you just need to test its association with user by using shoulda matchers.